pinot-core/src/test/java/org/apache/pinot/queries/MultiValueRawQueriesTest.java

```diff
@@ -338,6 +338,445 @@ public void testSelectQueries() {
+          + "ARRAYLENGTH(mvRawIntCol) < 5 LIMIT 10";
+      ResultTable resultTable = getBrokerResponse(query).getResultTable();
+      assertNotNull(resultTable);
+      DataSchema dataSchema = new DataSchema(new String[]{
+          "mvRawIntCol", "mvRawDoubleCol", "mvRawStringCol"
+      }, new DataSchema.ColumnDataType[]{
+          DataSchema.ColumnDataType.INT_ARRAY, DataSchema.ColumnDataType.DOUBLE_ARRAY,
+          DataSchema.ColumnDataType.STRING_ARRAY
+      });
+      assertEquals(resultTable.getDataSchema(), dataSchema);
+      List<Object[]> recordRows = resultTable.getRows();
+      assertEquals(recordRows.size(), 10);
+
+      for (int i = 0; i < 10; i++) {
+        Object[] values = recordRows.get(i);
+        assertEquals(values.length, 3);
+        int[] intVal = (int[]) values[0];
+        assertEquals(intVal[1] - intVal[0], MV_OFFSET);
+
+        double[] doubleVal = (double[]) values[1];
+        assertEquals(doubleVal[0], (double) intVal[0]);
+        assertEquals(doubleVal[1], (double) intVal[1]);
+
+        String[] stringVal = (String[]) values[2];
+        assertEquals(Integer.parseInt(stringVal[0]), intVal[0]);
+        assertEquals(Integer.parseInt(stringVal[1]), intVal[1]);
+      }
+    }
+    {
+      // Test a select with filter = query on an arraylength transform function
+      String query = "SELECT mvRawIntCol, mvRawDoubleCol, mvRawStringCol from testTable where "
+          + "ARRAYLENGTH(mvRawDoubleCol) = 2 LIMIT 10";
+      ResultTable resultTable = getBrokerResponse(query).getResultTable();
+      assertNotNull(resultTable);
+      DataSchema dataSchema = new DataSchema(new String[]{
+          "mvRawIntCol", "mvRawDoubleCol", "mvRawStringCol"
+      }, new DataSchema.ColumnDataType[]{
+          DataSchema.ColumnDataType.INT_ARRAY, DataSchema.ColumnDataType.DOUBLE_ARRAY,
+          DataSchema.ColumnDataType.STRING_ARRAY
+      });
+      assertEquals(resultTable.getDataSchema(), dataSchema);
+      List<Object[]> recordRows = resultTable.getRows();
+      assertEquals(recordRows.size(), 10);
+
+      for (int i = 0; i < 10; i++) {
+        Object[] values = recordRows.get(i);
+        assertEquals(values.length, 3);
+        int[] intVal = (int[]) values[0];
+        assertEquals(intVal[1] - intVal[0], MV_OFFSET);
+
+        double[] doubleVal = (double[]) values[1];
+        assertEquals(doubleVal[0], (double) intVal[0]);
+        assertEquals(doubleVal[1], (double) intVal[1]);
+
+        String[] stringVal = (String[]) values[2];
+        assertEquals(Integer.parseInt(stringVal[0]), intVal[0]);
+        assertEquals(Integer.parseInt(stringVal[1]), intVal[1]);
+      }
+    }
+    {
+      // Test a select with filter IN query on an arraylength transform function
+      String query = "SELECT mvRawIntCol, mvRawDoubleCol, mvRawStringCol from testTable where "
+          + "ARRAYLENGTH(mvRawStringCol) IN (2, 5) LIMIT 10";
```

Comment:
Is ORDER BY not needed to make the test result deterministic ?

Reply:
my initial checks only validated that the values across rows are as expected without validating explicitly what those values should be. I did update this to add an order by so that I could also validate explicit values